circuits/PointInPolygon.circom

```diff
@@ -0,0 +1,181 @@
+pragma circom 2.0.0;
+
+include "Utility.circom";
+include "Lines.circom";
+
+/*
+Returns 1 if a given point is in an n sided polygon, 0 otherwise.
+Implements ray tracing algorithm for simple polygons on a discrete 2^grid_bits length plane.
+Note, we don't check that the polygon is simple here, that must be done separately.
+*/
+template RayTracing(n, grid_bits) {
```

Comment:
thining of naming conventions, datatype~

Reply:
Yeah, so I don't think you can really have datatypes in circom. There are no structs or objects or anything. It's just parameters and signals, and those just have names.